cmd/crc/cmd/root.go

```diff
@@ -90,17 +91,32 @@ func runRoot() {
 	fmt.Println("No command given")
 }
 
+const (
+	defaultExitCode         = 1
+	preflightFailedExitCode = 2
+)
+
 func Execute() {
 	attachMiddleware([]string{}, rootCmd)
 
 	if err := rootCmd.ExecuteContext(telemetry.NewContext(context.Background())); err != nil {
 		runPostrun()
 		_, _ = fmt.Fprintln(os.Stderr, err.Error())
-		os.Exit(1)
+		var e *errorWithExitCode
+		if errors.As(err, &e) {
+			os.Exit(e.code)
+		} else {
+			os.Exit(defaultExitCode)
+		}
 	}
 	runPostrun()
 }
 
+type errorWithExitCode struct {
+	error
+	code int
+}
```

Comment:
I have 2 conflicting suggestions here :)
- https://pkg.go.dev/k8s.io/utils/exec#CodeExitError could be reused maybe?
- or `code` could be a separate type so that it's easier to find the corresponding constants if one needs them?

Reply:
I reused CodeExitError.